### PR TITLE
[Docs] Load Test #1 - Plan

### PR DESCRIPTION
**Pull request by Olshansk**
## Summary

Documentation for the goals of the first load test.

### Summary generated by Reviewpad on 05 Jan 24 17:00 UTC
This pull request includes a new file `docusaurus/docs/infrastructure/testing/_category_.json` that represents a category for testing. It contains a label of "Testing" with a position of 1, along with a link object that describes the type of end-to-end testing being done.

Additionally, a new document has been added to the repository, outlining the plan for the first load test for the Shannon upgrade. The document includes goals and non-goals for the load test, load profiles, measurements to be taken, architecture/component diagram, and tool requirements for the load test.

## Issue

- #229 

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [x] Documentation
- [ ] Other (specify)

### CODE REVIEW

File: docusaurus/docs/infrastructure/testing/load_test_1.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+| Block Publishing |      |      |   ❓    |      |
+| Rollup State     |      |      |         |   ❓ |
+
+#### 3. AppGate Server (Application, Gateway, etc…)
+
+`What`: Multiple `line charts` to capture `Disk` (size & iops), `RAM`, `CPU` , `Network` usage (ingress/egress)
+
+`Why`:
+
+- `Relay Proxies` Ingress/egress of relays could add
```
Comment (h5law) [inline docusaurus/docs/infrastructure/testing/load_test_1.md:130]: Could add what?

File: docusaurus/docs/infrastructure/testing/load_test_1.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+| Request Processing | ❓  | ❓  |         |      |
+| ???                |     |     |         |   ❓ |
+
+#### 4. RelayMiner (Supplier, SMT, etc..)
+
+`What`: Multiple `line charts` to capture `Disk` (size & iops), `RAM`, `CPU` , `Network` usage (ingress/egress)
+
+`Why`:
+
+- `SMT` - The `SMT` is one of the most important parts of the end-to-end flow which has impact on `RAM`, `CPU` & `disk`
```
Comment (h5law) [inline docusaurus/docs/infrastructure/testing/load_test_1.md:147]: > The `SMT` is one of the most important parts of the end-to-end flow

Hehe <3

File: docusaurus/docs/infrastructure/testing/load_test_1.md [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+  - [Script - Tools](#script---tools)
+
+## Goals
+
+1. `De-risk` the network’s feasibility to have completely permissionless services & actors
+   1. This is intended for `scalability` purposes and does not account for `Sybil` attacks
+2. `Stress test` the SMT (Sparse Merkle Trie) and/or how it is being used
+3. `Build intuition` into the cost of operating the network for all of the stakeholders involved, both on & off chain
+4. `Gain visibility` into basic metrics (disk, RAM, CPU, ingress/egress traffic, etc.…) for our network actors
+5. `Uncover` potential bugs or bottlenecks in both on-chain & off-chain code
```
Comment (red-0ne) [inline docusaurus/docs/infrastructure/testing/load_test_1.md:39]: We should also mention `potential concurrency issues`

File: docusaurus/docs/infrastructure/testing/load_test_1.md [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+
+#### 3. AppGate Server (Application, Gateway, etc…)
+
+`What`: Multiple `line charts` to capture `Disk` (size & iops), `RAM`, `CPU` , `Network` usage (ingress/egress)
+
+`Why`:
+
+- `Relay Proxies` Ingress/egress of relays could add
+- `Caches & State` - All the caching & state can have impact across the board
+- `Request Processing` - Signature generation, request marshaling / unmarshaling, etc…
```
Comment (red-0ne) [inline docusaurus/docs/infrastructure/testing/load_test_1.md:132]: Slow `Supplier` replies could increase the pending relays at the `AppGate` level (RAM)

**Review by red-0ne — COMMENTED** [review]
Awesome plan! Eager to contribute to it.

Left a couple of comments but nothing blocking.

Implicitly approving and let other reviewers hit the approve button :)

File: docusaurus/docs/infrastructure/testing/load_test_1.md [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+    A <-- Req/Res --> G
+    G <-- Req/Res --> S
+
+    P -. metrics .-> M
+    G -. metrics .-> M
+    S -. metrics .-> M
+
+    classDef orangeColor fill:#FF8C00;
+    classDef purpleColor fill:#800080;
+    classDef blueColor fill:#0000FF;
```
Comment (h5law) [inline docusaurus/docs/infrastructure/testing/load_test_1.md:224]:
```suggestion
    classDef blueColor fill:#75A8E6;
```
To me this was pink

File: docusaurus/docs/infrastructure/testing/load_test_1.md [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+
+    A <-- Req/Res --> G
+    G <-- Req/Res --> S
+
+    P -. metrics .-> M
+    G -. metrics .-> M
+    S -. metrics .-> M
+
+    classDef orangeColor fill:#FF8C00;
+    classDef purpleColor fill:#800080;
```
Comment (h5law) [inline docusaurus/docs/infrastructure/testing/load_test_1.md:223]:
```suggestion
    classDef purpleColor fill:#B160E3;
```
Hopefully this makes the text more readable

File: docusaurus/docs/infrastructure/testing/load_test_1.md [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+
+## Tool Requirements
+
+### Deployment Environment
+
+- Ability to deploy the environment (and tooling) on LocalNet & DevNet
+
+### Request Source / Generator
+
+- A script/tool to generate N requests per second
```
Comment (h5law) [inline docusaurus/docs/infrastructure/testing/load_test_1.md:247]:
```suggestion
- A script/tool to generate $N$ requests per second
```

**Review by h5law — CHANGES_REQUESTED** [review]
Looks good, had a lot of trouble with the diagram left some suggestions there. Overall nice plan LFG!

**Review by red-0ne — APPROVED** [review]
LGTM

**Pull request merged**